client/http-client/src/transport.rs

```diff
@@ -258,6 +260,20 @@ impl<L> HttpTransportClientBuilder<L> {
 			}
 		}
 
+		match url.password() {
+			Some(pwd) if !url.username().is_empty() => {
```

Comment:
Does CURL also not set anything if no username is given?

Reply:
Tested right now, curl allow empty username let's do the same then

Reply:
remove this check